easybuild/easyblocks/e/elpa.py

```diff
@@ -0,0 +1,227 @@
+                    setattr(self, flag, True)
+
+    def run_all_steps(self, *args, **kwargs):
+        """
+        Put configure options in place for different builds (with and without openmp).
+        """
+
+        # make all builds verbose
+        self.cfg.update('preconfigopts', 'autoreconf &&')
+
+        # the following configopts are common to all builds
+        if self.toolchain.options['pic']:
+            self.cfg.update('configopts', '--with-pic')
+
+        if self.cfg['with_shared']:
+            self.cfg.update('configopts', '--enable-shared')
+
+        if self.cfg['with_generic_kernel']:
+            self.cfg.update('configopts', '--enable-generic')
+
+        if self.cfg['with_single']:
+            self.cfg.update('configopts', '--enable-single-precision')
+
+        for flag in ELPA_CPU_FEATURE_FLAGS:
+            # many ELPA kernels are enabled by default, even when the
+            # CPU does not support them, so we disable them all, except
+            # when the appropriate CPU flag is found
+            # sse kernels require sse4_2
+            if flag == 'sse4_2':
+                if getattr(self, flag):
+                    self.cfg.update('configopts','--enable-sse')
+                    self.cfg.update('configopts', '--enable-sse-assembly')
+                else:
+                    self.cfg.update('configopts', '--disable-sse')
+                    self.cfg.update('configopts', '--disable-sse-assembly')
+            elif flag == 'avx512f':
+                if getattr(self, 'avx512f'):
+                    self.cfg.update('configopts', '--enable-avx512')
+                else:
+                    self.cfg.update('configopts', '--disable-avx512')
+            else:
+                if getattr(self, flag):
+                    self.cfg.update('configopts', '--enable-%s' % flag)
+                else:
+                    self.cfg.update('configopts', '--disable-%s' % flag)
+
+        # By default ELPA tries to use MPI and configure fails if it's not available
+        # so we turn it off if MPI is not available
+        # We also set the LIBS environmet variable to detect the correct linalg library
+        # depending on the MPI availability.
+        if self.toolchain.options.get('usempi', None):
+            self.cfg.update('configopts', 'LIBS="$LIBSCALAPACK"')
+        else:
+            self.cfg.update('configopts', '--with-mpi=no')
+            self.cfg.update('configopts', 'LIBS="$LIBLAPACK"')
+
+        # make all builds verbose
+        self.cfg.update('buildopts', 'V=1')
+
+        # keep track of common configopts and of configopts specified
+        # in easyconfig file, so we can include them in each iteration
+        # later
+        common_config_opts = self.cfg['configopts']
+        common_build_opts = self.cfg['buildopts']
+
+        self.cfg['configopts'] = []
+        self.cfg['buildopts'] = []
```

Comment:
There's no need to also make `buildopts` a list here, since the build options are always the same.

If `buildopts` is just a string, the same value will be used for every iteration by default.

Less code to worry about :)

Reply:
Yes, you are right. Those are "leftovers" from the previous version where we were also looping over MPI on/off.